clients/drcachesim/analysis_tool.h

```diff
@@ -167,35 +167,31 @@ template <typename RecordType> class analysis_tool_tmpl_t {
      * framework automatically.
      */
     struct interval_state_snapshot_t {
-        interval_state_snapshot_t(int shard_id, uint64_t interval_id,
-                                  uint64_t interval_end_timestamp)
-            : shard_id_(shard_id)
-            , interval_id_(interval_id)
-            , interval_end_timestamp_(interval_end_timestamp)
+        interval_state_snapshot_t(uint64_t interval_id, uint64_t interval_end_timestamp)
+            : interval_id(interval_id)
+            , interval_end_timestamp(interval_end_timestamp)
         {
         }
         interval_state_snapshot_t()
-            : shard_id_(0)
-            , interval_id_(0)
-            , interval_end_timestamp_(0)
+            : interval_id(0)
+            , interval_end_timestamp(0)
         {
         }
 
-        int shard_id_;
```

Comment:
If a tool wants to do something like that, we'd need to figure out another way to tell the tool about the shard-id. Today the user returned `shard_data` pointer is the only way.

Or maybe if a tool wants to do this, it can add its own identifiers inside the snapshot struct, like a tid or something.

Reply:
(See other comment: this shard_id is an analyzer-internal id, never communicated to the tool)